src/libraries/System.Diagnostics.Process/src/Resources/Strings.resx

```diff
@@ -185,6 +185,9 @@
   <data name="InvalidApplication" xml:space="preserve">
     <value>The specified executable is not a valid application for this OS platform.</value>
   </data>
+  <data name="FailedToStartFileDirectory" xml:space="preserve">
+    <value>Failed to start '{0}' with working directory '{1}'. {2}</value>
```

Comment:
It's just an opinion (and I can see that it was a suggestion in the original issue https://github.com/dotnet/runtime/issues/39928#issuecomment-664046020), but I think that it would be better to append the missing information to the existing exception message. Especially if we want to add it for all throwing cases (not just when the file is not found)

```suggestion
    <value>{2} (the file name was '{0}', the working directory '{1}').</value>
```

So we would have:

```diff
- The system cannot find the file specified
+ The system cannot find the file specified (the file name was 'abc.exe', the working directory 'C:\xyz').
```

Instead of:

```diff
- The system cannot find the file specified
+ Failed to start 'abc.exe' with working directory 'C:\xyz'. The system cannot find the file specified
```

cc @danmosemsft who proposed the initial version

Reply:
How about "{0} (file: '{1}', working directory: '{2}')." then?

Given that "file name" is not always appropriate as it could also be an absolute/relative path to some exe.
Even file is not always the right term/word as it could also be "ftp://something" or "https://something", but "file" sounds better than "file name" here.
Or should we stick with "file name" as the property is also called "FileName"?

Are there native errors that could produce an empty message?
In that case the original proposed message would still make sense whereas the new one would look awkward. 

```diff
- Failed to start 'abc.exe' with working directory 'C:\xyz'. 
+  (the file name was 'abc.exe', the working directory 'C:\xyz').
```

Reply:
> How about "{0} (file: '{1}', working directory: '{2}')." then?

Sounds good to me! 

> Are there native errors that could produce an empty message?

This is a very good question. From what I can see in the source code, there is no such possibility for Windows:

https://github.com/dotnet/runtime/blob/144cac3d0ce162444452a5dcc837b904fe149ef4/src/libraries/Common/src/Interop/Windows/Kernel32/Interop.FormatMessage.cs#L74

And on Unix we end up calling `strerror_r`:

https://github.com/dotnet/runtime/blob/144cac3d0ce162444452a5dcc837b904fe149ef4/src/libraries/Common/src/Interop/Unix/Interop.Errors.cs#L170

https://github.com/dotnet/runtime/blob/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/libraries/Native/Unix/System.Native/pal_errno.c#L18

and the [docs](https://man7.org/linux/man-pages/man3/strerror.3.html) for it says that it also returns unknow error message for such cases:

>  The strerror(), strerror_l(), and the GNU-specific strerror_r()
       functions return the appropriate error description string, or an
       "Unknown error nnn" message if the error number is unknown.

So I think that we can assume that it's never empty.

Reply:
With respect to the message, my only thought is whether exceptions might occur that wouldn't suggest we were trying to start a process. Eg., `System is not in a good state (the file name was 'abc.exe', the working directory 'C:\xyz').` (I made this up - but OS sometimes give strange errors). If you don't have the callstack, only the message, it is not clear what is going on. This problem already exists of course - but that was my reasoning for the "failed to start" text.
But this is @adamsitnik call.

Reply:
After some manual testing, such as where the `filename` provided is a directory, the message, "No such file or directory (file: '/home`, working directory: '/home')" seems confusing to me. I agree with the previous comments that having "failed to start" at the beginning is a better experience; the second part of the message is the underlying error.

In that spirit, I'll go with "An error occurred trying to start process '{0}' with working directory '{1}'. {2}".